txscript/scriptnum.go

```diff
@@ -164,11 +168,16 @@ func (n scriptNum) Int32() int32 {
 // [0x7f 0x00 0x00 ...], etc.  All forms except [0x7f] will return an error with
 // requireMinimal enabled.
 //
+// The scriptNumLen is the size of the data to be encoded.
```

Comment:
This isn't quite right.  The number is already encoded when this function is called.  Perhaps the following?

```Go
// The scriptNumLen is the maximum number of bytes the encoded value can be
// before an ErrStackNumberTooBig is returned.  This effectivley limits the
// range of allowed values.
// WARNING:  Great care should be taken if passing a value larger than
// defaultScriptNumLen, since it could lead to addition and multiplication
// overflows.
//
```